packages/astro/src/types/public/config.ts

```diff
@@ -1541,7 +1541,25 @@ export interface AstroUserConfig {
 	 * These flags allow you to opt in to some deprecated or otherwise outdated behavior of Astro
 	 * in the latest version, so that you can continue to upgrade and take advantage of new Astro releases.
 	 */
-	legacy?: object;
+	legacy?: {
+		/**
+		 * @docs
+		 * @name legacy.collections
+		 * @type {boolean}
+		 * @default `false`
+		 * @version 5.0.0
+		 * @description
+		 * Enable legacy behavior for content collections.
+		 * If enabled, data and content collections are handled using the legacy code instead of Content Layer API. 
+		 * Any collections with a loader defined will still use the Content Layer API. When enabled, you cannot use 
+		 * the glob loader for any collections in the `src/content` directory, and they will instead be handled by
+		 * the legacy code.
+		 *
+		 * To migrate to the new Content Layer API, you should remove this flag and define a collection for any 
+		 * directories in `src/content` that you want to use as a collection.
```

Comment:
```suggestion
		 * Enable legacy behavior for content collections.
		 * 
		 * ```js
		 * // astro.config.mjs
		 * import { defineConfig } from 'astro/config';
		 * export default defineConfig({
		 *   legacy: {
		 *     collections: true
		 *   }
		 * });
		 * ```
		 *
		 * If enabled, `data` and `content` collections (only) are handled using the legacy content collections implementation. Collections with a `loader` (only) will continue to use the Content Layer API instead. Both kinds of collections may exist in the same project, each using their respective implementations.
		 *  
		 *  The following limitations continue to exist:
		 *
		 * - Any legacy (`type: 'content'` or `type: 'data'`) collections must continue to be located in the `src/content/` directory.
		 * - These legacy collections will not be transformed to implicitly use the `glob()` loader, and will instead be handled by legacy code.
		 * - Collections using the Content Layer API (with a `loader` defined) are forbidden in `src/content/`, but may exist anywhere else in your project. 
		 *
		 * When you are ready to remove this flag and migrate to the new Content Layer API for your legacy collections, you must define a collection for any directories in `src/content/` that you want to continue to use as a collection. It is sufficient to declare an empty collection, and Astro will implicitly generate an appropriate definition for your legacy collections:
		 *  
		 * ```js
		 * // src/content/config.ts
		 * import { defineCollection, z } from 'astro:content';
		 * 
		 * const blog = defineCollection({ })
		 *  
		 * export const collections = { blog };
		 * ```
		 *

```
It's everyone's least favourite game: write that JSDoc! I'm thinking maybe something like this is helpful? What do you think?